### DISCUSSION TITLE
Report Manager Error

### DISCUSSION THREAD

**Discussion by feetpeoplellc** [abuzuhri/Amazon-SP-API-CSharp#114] [Q&A]
I apologize for my ignorance, in advance, but I am having difficulty getting this to work.  I have the below code, so not many changes.  I have changed the keys below to represent the correct credentials.  I was originally getting an authentication error, but that is resolved, so I "think" it is connecting properly now.  The MarketPlace has been changed to the US.

`AmazonConnection amazonConnection = new AmazonConnection(new AmazonCredential()
{
     AccessKey = "AKIAXXXXXXXXXXXXXXX",
     SecretKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     RoleArn = "arn:aws:iam::XXXXXXXXXXXXX:role/XXXXXXXXXXXX",
     ClientId = "amzn1.application-XXX-client.XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     ClientSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     RefreshToken= "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
     MarketPlace = MarketPlace.GetMarketPlaceByID("ATVPDKIKX0DER")
});

ReportManager reportManager = new ReportManager(amazonConnection);
var products = reportManager.GetProducts();
`

When executing the above, I get an error message:
FikaAmazonAPI.AmazonSpApiSDK.Models.Exceptions.AmazonException
  HResult=0x80131500
  Message=Amazon Api didn't respond with Okay, see exception for more details
  Source=FikaAmazonAPI
  StackTrace:
   at FikaAmazonAPI.Services.RequestService.ParseResponse(IRestResponse response)
   at FikaAmazonAPI.Services.RequestService.ExecuteRequest[T]()
   at FikaAmazonAPI.Services.ReportService.CreateReport(ParameterCreateReportSpecification createReportSpecification)
   at FikaAmazonAPI.Services.ReportService.CreateReportAndDownloadFile(ReportTypes reportTypes, Nullable`1 dataStartTime, Nullable`1 dataEndTime, ReportOptions reportOptions)
   at FikaAmazonAPI.ReportGeneration.ReportManager.GetProducts(AmazonConnection amazonConnection)
   at FikaAmazonAPI.ReportGeneration.ReportManager.GetProducts()
   at Amazon_Reports__SP_API_.Program.Main(String[] args) in C:\Users\xxxxx\source\repos\Amazon Reports (SP-API)\Amazon Reports (SP-API)\Program.cs:line 45

Line 45 is:
`var products = reportManager.GetProducts();`

One more point...  I also tried the next example of getting a report that has a bit more code.  However, I received the same error message as above (Amazon Api didn't respond with Okay).

Thank you for your time and any assistance as I am sure it is something that I am not doing properly.

**Comment by abuzuhri** [comment]
Hello @feetpeoplellc
Is the other methods work with you ?

**Comment by feetpeoplellc** [comment]
Thank you for the quick response.  I am getting the same error for the "Orders List with Parameter" example also.  I will double check permissions once again, as this seems to be the commonality at this point.

**Reply by abuzuhri** [reply]
Yes please recheck key and role , look like the problem there

**Reply by feetpeoplellc** [reply]
I am marking this issue resolved.  As expected, it was something on our end that was causing the issue.  We did not start the project "clean" and had extra nuget packages loaded, so perhaps one or more of these had conflicted with the solution and would not allow it to execute.  When we started a fresh project and only loaded this solution from Nuget, it worked perfectly.

This can be closed.
